### PR TITLE
Update UpsertTracepoint and bpftrace deployment documentation

### PR DESCRIPTION
**Pull request by benkilimnik**
Summary: Documents the new `TraceProgram` object added to `pxtrace` which enables users to specify deployment restrictions for BPFtrace programs. Changes introduced in https://github.com/pixie-io/pixie/pull/1687. Should not be merged until those changes land.

Related issues/PRs: https://github.com/pixie-io/pixie/issues/1659

### CODE REVIEW

File: external/pxl_documentation.json [hunk 1]
```diff
@@ -1045,9 +1045,9 @@
...
           {
             "ident": "probe_fn",
-            "desc": "The tracepoint function.",
+            "desc": "The tracepoint function, BPFTrace program or pxtrace.TraceProgram to deploy.",
             "types": [
-              "px.ProbeFn"
+              "Union[px.ProbeFn, str, pxtrace.TraceProgram, List[pxtrace.TraceProgram]]"
             ]
           },
           {
```
Comment (ddelnano) [inline external/pxl_documentation.json]: @benkilimnik can you confirm if that was hand edited? This pxl documentation file will be automatically updated when the next vizier release happens ([source](https://github.com/pixie-io/pixie/blob/52e003d5ad64aca400ec701c017e3dcfe462f574/.github/workflows/release_update_docs_px_dev.yaml)).

So we should make sure the change above is modified [here](https://github.com/pixie-io/pixie/blob/52e003d5ad64aca400ec701c017e3dcfe462f574/src/carnot/planner/probes/tracing_module.h#L193) and then the next release will update the file for us.

In addition to that, it _should_ add the [pxtrace.TraceProgram documentation](https://github.com/pixie-io/pixie/pull/1687/files#diff-5f51cf15f400e00e06345ed177be7d130567789b740bc3ff2f168a869c103aacR200), however, it seems we are missing a call to `SetDocString`.
```
# See that the new doc string isn't referenced
$ git grep kTraceProgramDocstring
src/carnot/planner/probes/tracing_module.h:  inline static constexpr char kTraceProgramDocstring[] = R"doc(

# See a working example calls ::SetDocString
$ git grep kDeleteTracepointDocstring
src/carnot/planner/probes/tracing_module.cc:  PX_RETURN_IF_ERROR(delete_fn->SetDocString(kDeleteTracepointDocstring));
src/carnot/planner/probes/tracing_module.h:  inline static constexpr char kDeleteTracepointDocstring[] = R"doc(
```

So it seems like we need another PR that includes both of those changes.

Comment (benkilimnik) [reply external/pxl_documentation.json]: Just opened [this PR](https://github.com/pixie-io/pixie/pull/1720) which updates the `probe_fn` description and adds the call to `SetDocString`.

**Review by ddelnano — APPROVED** [review]

**Comment by benkilimnik** [comment]
@pixie-io/admin this is ready for review.

**Comment by ddelnano** [comment]
Also wanted to note that the necessary vizier changes are included in the current release ([v0.14.7](https://github.com/pixie-io/pixie/releases/tag/release%2Fvizier%2Fv0.14.7)).

**Review by aimichelle — APPROVED** [review]

**Pull request merged**